### PR TITLE
Feature/bug hunting

### LINKED ISSUE #52: Refactor getFullRoomInfoForBookings
Not great to modify properties on the customer class directly from the hotel class like this:

customer.bookings.push(roomsBooked);

### LINKED ISSUE #53: Re-examine hotel.getUserExpenses
Might move this to the customer class and then just call hotel.getFullRoomInforForBookings() inside of that function

While we're at it, that is a wildly long method name

### LINKED ISSUE #57: Remove number of guests from HTML unless you're gonna use it to filter the search

### LINKED ISSUE #64: UI for dashboard if a user has no upcoming stays
Earline Hamill doesn't have any!

### LINKED ISSUE #68: bugfix: should not be able to log in with username03

### LINKED ISSUE #69: UI- decision, what should happen when you just enter "user" as username, what message?

### LINKED ISSUE #70: UX- should you go to your dashboard after a successful booking or back to the landing page?

### LINKED ISSUE #71: UI total expenses- should they be at top of dashboard instead of bottom?

### PR DESCRIPTION
**Pull request by clairefields15**
# Description
UI/UX improvements: removed 'number of guests' field, moved total expenses to top of user dashboard, added logic to display a message in dashboard if you have no upcoming trips, after booking a room redirect to dashboard instead of landing page

Bug fix: should not be able to log in with username03, added logic if id starts with 0

Refactoring: added setBookings as a customer method so that hotel is no longer modifying the this.bookings property of the customer. Also added getExpenses as a customer method and added a property of this.expenses since it makes more sense from a data flow perspective.

## Issues Closed
closes #57 
closes #52 
closes #71 
closes #53
closes #64
closes #70
closes #69
closes #68

## Type of change
Check the relevant option:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (clean up code)

## How Has This Been Tested?
running both servers

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -224,8 +224,26 @@ const domUpdates = {
...
+      dayjs(booking.booking.date).isSame(currentDate)
+    );
+
+    if ((findUpcomingBookings.length + findPresentBookings.length) === 0) {
+      upcomingStays.innerHTML += `
+        <article class="past-booking-card">
+          <p>You have no upcoming stays.</p>
+        </article>
+      `;
+    }
```
Comment (jon-schlandt) [inline src/domUpdates.js]: Hey, nice work implementing some error handling here! It's a good idea to give the users some sort of message if no information is displaying on a page.

I see that this logic, and subsequent rendering, is within a larger 'renderUserDashboard()' method. Could this be taken out and and encapsulated into a smaller function to increase SRP?

File: src/scripts.js [hunk 2]
```diff
@@ -180,14 +173,16 @@ function checkEmptyFields(username, password) {
...
-  passwordField.value = '';
 
-  // this might be why username03 can log in as if username3
   let id = username.slice(8);
-  if (id) {
+  if (id.startsWith(0)) {
+    domUpdates.showLogInError();
+    errorTag.innerText = 'User does not exist, please try again.';
+    throw new Error('User does not exist');
+  } else if (id) {
```
Comment (jon-schlandt) [inline src/scripts.js:185]: Good work finding a solution to this issue! When the error is thrown, are you responding to it with a message on the DOM?

File: src/customer.js [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+
+  setBookings(hotel) {
+    this.bookings = hotel.getFullRoomInfoForBookings(this)
+    return this.bookings
+  }
+
+  getExpenses(hotel) {
+    this.expenses = hotel.getUserExpenses(this) 
+    return this.expenses
+  }
```
Comment (jon-schlandt) [inline src/customer.js:18]: It looks like the jobs of these methods are to call upon methods from the Hotel class and return their results - is there a way to simply call the methods directly from Hotel instead?   

Unless you are using ‘this.bookings’ and ‘this.expenses’ in other places within Customer, which I could see the benefit of holding onto this info.

**Review by jon-schlandt — COMMENTED** [review]
I Really like the error handling here - good work thinking about the UX of your app! I left a few comments with questions that came up when reading over your changes.

File: src/domUpdates.js (hunk 1 above)
Comment (clairefields15) [reply src/domUpdates.js]: Yes definitely! This function is huuuuge at the moment and has a lot of logic in it. It's on my refactoring list.

File: src/scripts.js (hunk 2 above)
Comment (clairefields15) [reply src/scripts.js:185]: Yes, but I'd like to handle it better, it is showing a separate screen with the error for 4 seconds and then going back to the log in screen

**Pull request merged**